src/miner.cpp

```diff
@@ -134,45 +172,49 @@ class AddOutputsToCoinbaseTxAndSign : public boost::static_visitor<>
         const int nHeight,
         const CAmount nFees) : mtx(mtx), chainparams(chainparams), nHeight(nHeight), nFees(nFees) {}
 
-    CAmount SetFoundersRewardAndGetMinerValue() const {
-        auto value = GetBlockSubsidy(nHeight, chainparams.GetConsensus());
-
-        if ((nHeight > 0) && (nHeight <= chainparams.GetConsensus().GetLastFoundersRewardBlockHeight(nHeight))) {
-            // Founders reward is 20% of the block subsidy
-            auto vFoundersReward = value / 5;
-            // Take some reward away from us
-            value -= vFoundersReward;
-            // And give it to the founders
-            mtx.vout.push_back(CTxOut(vFoundersReward, chainparams.GetFoundersRewardScriptAtHeight(nHeight)));
+    const libzcash::Zip212Enabled GetZip212Flag() const {
+        if (chainparams.GetConsensus().NetworkUpgradeActive(nHeight, Consensus::UPGRADE_CANOPY)) {
+            return libzcash::Zip212Enabled::AfterZip212;
+        } else {
+            return libzcash::Zip212Enabled::BeforeZip212;
         }
-
-        return value + nFees;
     }
 
-    void operator()(const InvalidMinerAddress &invalid) const {}
-
-    // Create shielded output
-    void operator()(const libzcash::SaplingPaymentAddress &pa) const {
-        auto value = SetFoundersRewardAndGetMinerValue();
-        mtx.valueBalance = -value;
-
-        uint256 ovk;
-        libzcash::Zip212Enabled zip_212_enabled = libzcash::Zip212Enabled::BeforeZip212;
-        if (Params().GetConsensus().NetworkUpgradeActive(nHeight, Consensus::UPGRADE_CANOPY)) {
-            zip_212_enabled = libzcash::Zip212Enabled::AfterZip212;
+    CAmount SetFoundersRewardAndGetMinerValue(void* ctx) const {
+        auto miner_reward = GetBlockSubsidy(nHeight, chainparams.GetConsensus());
```

Comment:
The main reason I hadn't written it this way was that the full block subsidy isn't otherwise used. So miner_reward is initialized to the block subsidy, then decremented.

Reply:
Well it kind-of is used: the second parameter to `GetActiveFundingStreamElements` is the block subsidy, not the miner reward. (It works currently because `miner_reward` is still equal to the block subsidy at that call, but I think it would be clearer and less fragile under maintenance the way I suggest.)